cloudblue_connect.custom_adapter.rb

```diff
@@ -0,0 +1,694 @@
+            label: "Customer State Or Province",
+            hint: "",
+            optional: false
+          },
+          
+              {
+            name: "customer_country",
+            label: "Customer Country",
+            hint: "Provide the country of the related account, please use Alpha-2 code.",
+            optional: false
+          },
+          
+              {
+            name: "customer_contact_first_name",
+            label: "Customer Contact First Name",
+            hint: "Provide the contact name of the technical contact related with the account.",
+            optional: false
+          },
+          
+              {
+            name: "customer_contact_last_name",
+            label: "Customer Contact Last Name",
+            hint: "Provide the last name or family name of the technical contact related with the account.",
+            optional: false
+          },
+          
+                {
+            name: "customer_email",
+            label: "Customer Email",
+            hint: "Provide the email address of the technical contact related with the account, this one typically is a requirement by vendors to process the request, additionally please note that this email will be used to send inquire requests related with the fulfillment of the request.
+",
+            optional: false
+          },
+          
+                {
+            name: "customer_phone_number",
+            label: "Customer Phone Number",
+            hint: "Provide the phone number of the technical contact. We suggest to use international phone notation in order to ensure the correctness of the phone number.
+
+",
+            optional: true
+          },
+          
+          
+                {
+            name: "product_id",
+            label: "Product ID",
+            hint: "Provide the Product's ID.
+
+            ",
+            optional: false
+          },
+          
+          {
+            name: "product_items",
+            label: "Product Items",
+            hint: "",
+            optional: false,
+            type: "array", of: "object", properties: [
+            { optional: false, name: "id", label: "Item ID" },
+            { optional: false, name: "quantity", label: "Quantity" }
+            ]
+          },
+          
+          
+        ]
+      end,
+      
+      execute: lambda do |connection, input|
+        
+        x = 4
+        rand_uid_num = (rand(10 ** x) + 10**x).to_s[0..x-1]
+        rand_uid_num_two = (rand(10 ** x) + 10**x).to_s[0..x-1]
+        rand_id_num = (rand(10 ** x) + 10**x).to_s[0..x-1]
+        first_item = input["product_items"][0]
+        #"id"=>"#{first_item["id"].slice(0, first_item["id"].rindex("-"))}"
+        hardcoded_params = {
+           "type"=>"purchase",
+           "asset"=>{
+           "product"=>{
+           "id"=>"#{input["product_id"]}"
+             
+      },
+      "connection"=>{
+         "id"=>"CT-0000-0000-0000",
+         "type"=>"production"
+      },
```

Comment:
I think we should omit this. Connect will derive connection from product and hub/marketplace.